tests/test_atoms_state.py

```diff
@@ -0,0 +1,350 @@
+                0.5,
+                None,
+                (0.5, None, 1.5),
+            ),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'initial', (None, 1, None)),
+            (OrbitalsState(l_quantum_number=1), 0.5, 'final', (0.5, None, 5.5)),
+            (None, 0.5, None, (0.5, None, None)),
+        ],
+    )
+    def test_normalize(self, orbital_ref, n_excited_electrons, dscf_state, results):
+        """
+        Test the normalization of the `CoreHole`. Inputs are defined as the quantities of the `CoreHole` section.
+        """
+        core_hole = CoreHole()
+        core_hole.orbital_ref = orbital_ref
+        core_hole.n_excited_electrons = n_excited_electrons
+        core_hole.dscf_state = dscf_state
+
+        core_hole.normalize(None, logger)
+
+        assert core_hole.n_excited_electrons == results[0]
+        if core_hole.orbital_ref:
+            assert core_hole.orbital_ref.degeneracy == results[1]
+            assert core_hole.orbital_ref.occupation == results[2]
+
+
+class TestHubbardInteractions:
+    """
+    Test the `HubbardInteractions` class defined in atoms_state.py.
+    """
+
+    @staticmethod
+    def add_u_j(hubbard_interactions, u, j) -> None:
+        """Adds `u_interaction` and `j_local_exchange_interaction` (in eV) to the `HubbardInteractions` section."""
+        if u is not None:
+            hubbard_interactions.u_interaction = u * ureg('eV')
+        if j is not None:
+            hubbard_interactions.j_local_exchange_interaction = j * ureg('eV')
+
+    @pytest.mark.parametrize(
+        'slater_integrals, results',
+        [
+            ([3.0, 2.0, 1.0], (0.1429146, -0.0357286, 0.0893216)),
+            (None, (None, None, None)),
+            ([3.0, 2.0, 1.0, 0.5], (None, None, None)),
+        ],
+    )
+    def test_u_interactions(self, slater_integrals, results):
+        """
+        Test the Hubbard interactions `U`, `U'`, and `J` for a given set of Slater integrals.
+        """
+        # Adding `slater_integrals` to the `HubbardInteractions` section
+        hubbard_interactions = HubbardInteractions()
+        if slater_integrals is not None:
+            hubbard_interactions.slater_integrals = slater_integrals * ureg('eV')
+
+        # Resolving U, U', and J from class method
+        (
+            u_interaction,
+            u_interorbital_interaction,
+            j_hunds_coupling,
+        ) = hubbard_interactions.resolve_u_interactions(logger)
+
+        if None not in (u_interaction, u_interorbital_interaction, j_hunds_coupling):
+            assert np.isclose(u_interaction.to('eV').magnitude, results[0])
+            assert np.isclose(u_interorbital_interaction.to('eV').magnitude, results[1])
+            assert np.isclose(j_hunds_coupling.to('eV').magnitude, results[2])
+        else:
+            assert (
+                u_interaction,
+                u_interorbital_interaction,
+                j_hunds_coupling,
+            ) == results
```

Comment:
How about you define your own comparison that combines `np.isclose` and `==`?
Alternatively, I remember older tests using `approximate` from `pytest`.